### PR TITLE
Improve IBM CP4D installation on IBM Cloud

### PR DESCRIPTION
**Pull request by jensmueller-com**
This pull request contains the following changes:

-  Add CLI option to specify an alias when creating a Red Hat OpenShift cluster on IBM Cloud
-  Add CLI option to specify an OpenShift version when creating a Red Hat OpenShift cluster on IBM Cloud
-  Always log in to IBM Cloud when creating a Red Hat OpenShift cluster on IBM Cloud as `dg.lib.ibmcloud.login.is_logged_in()` returns `True` even if `user` key is returned by `ibmcloud target --output json`, but IAM access token expired
-  Add Red Hat OpenShift cluster on IBM Cloud cluster to `~/.dg/clusters.json` after having created it
-  Remove Red Hat OpenShift cluster on IBM Cloud from `~/.dg/clusters.json` after having deleted it
-  Replace `dg.lib.ibmcloud.plugin` module (function-based) with `dg.lib.ibmcloud.plugin_manager` module (class-based)
-  Replace `dg.lib.ibmcloud.vlan` module (function-based) with `dg.lib.ibmcloud.vlan_manager` module (class-based)

### CODE REVIEW

File: dg/lib/ibmcloud/vlan_manager.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        )
+
+        self._default_public_vlan = self._get_default_vlan_id(
+            ibmcloud_oc_vlan_ls_command_result,
+            ibmcloud_oc_vlan_ls_command_result_json,
+            "public",
+            zone,
+        )
+
+    def get_default_private_vlan(self) -> str:
```
Comment (johker01) [inline dg/lib/ibmcloud/vlan_manager.py:38]: as this is a getter, shouldn't you use `@property` and name the method itself `default_private_vlan`?
See e.g. https://stackoverflow.com/questions/6618002/using-property-versus-getters-and-setters.

File: dg/lib/ibmcloud/vlan_manager.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        return self._default_private_vlan
+
+    def get_default_public_vlan(self) -> str:
```
Comment (johker01) [inline dg/lib/ibmcloud/vlan_manager.py:41]: same as above

**Review by johker01 — APPROVED** [review]
I only had some minor comments. Otherwise, the PR looks good to me.

File: dg/commands/ibmcloud/oc/cluster/create.py [hunk 3]
```diff
@@ -14,31 +14,58 @@
...
-    get_default_private_vlan,
-    get_default_public_vlan,
-)
+from dg.lib.ibmcloud.vlan_manager import VLANManager
 from dg.utils.logging import loglevel_command
 
 logger = logging.getLogger(__name__)
 
 
+def validate_ocp_version(ctx, param, value) -> Optional[semver.VersionInfo]:
```
[the remark this answers is not in the document]
Comment (jensmueller-com) [reply dg/commands/ibmcloud/oc/cluster/create.py:47]: No, this is the callback signature required by [Click](https://click.palletsprojects.com/en/7.x/options/#callbacks-and-eager-options).

**Pull request merged**